src/states/covenant.ts

```diff
@@ -86,7 +86,7 @@ export class Covenant {
         if (isNumber(covenant)) {
             return ReturnBoolean(this.covenantId === covenant);
         } else if (isString(covenant)) {
-            const name = tostring(covenant);
+            const name = covenant as string;
```

Comment:
This should not be necessary

Reply:
Hmm, when I originally submitted the changes to `covenant.ts`, the check bot on GitHub complained about errors during the compile phase, but when I run them locally now, the errors aren't there.  I will change this a bit more.